libs/common-services/src/main/java/uk/gov/di/ipv/core/library/service/IpvSessionService.java

```diff
@@ -174,4 +177,23 @@ public void updateIpvSession(IpvSessionItem updatedIpvSessionItem) {
     private String toExpiryDateTime(long expirySeconds) {
         return Instant.now().plusSeconds(expirySeconds).toString();
     }
+
+    private <T> T callRunTaskWithBackoff(RetryableTask<T> task)
+            throws GetIpvSessionException, UnknownAccessTokenException {
+        try {
+            return Retry.runTaskWithBackoff(sleeper, 7, 10, task);
+        } catch (InterruptedException e) {
+            LOGGER.warn(LogHelper.buildLogMessage("backoff and retry sleep was interrupted"));
+            Thread.currentThread().interrupt();
+            throw new GetIpvSessionException("Failed to get ipv session", e);
+        } catch (NonRetryableException e) {
+            LOGGER.warn(
+                    LogHelper.buildErrorMessage(
+                            "exception occurred while retrying to get ipv session", e));
+            if (e.getCause() instanceof UnknownAccessTokenException uatException) {
```

Comment:
Should we have a similar block here for `e.getCause() instanceof GetIpvSessionException`?